src/main/java/com/Elessar/database/MongoDB.java

```diff
@@ -0,0 +1,110 @@
+package com.Elessar.database;
+
+import com.Elessar.app.server.User;
+import com.mongodb.client.MongoCursor;
+import com.mongodb.client.MongoDatabase;
+import org.bson.BsonDocument;
+import org.bson.Document;
+import org.bson.conversions.Bson;
+import java.util.ArrayList;
+import java.util.List;
+import static com.mongodb.client.model.Filters.and;
+import static com.mongodb.client.model.Filters.eq;
+
+/**
+ * Created by Hans on 1/20/19.
+ */
+public class MongoDB implements MyDatabase{
+    private final MongoDatabase db;
+    public MongoDB(MongoDatabase db) {
+        this.db = db;
+    }
+
+    @Override
+    public void insert(User user) {
+        final Document doc = new Document();
+        doc.append(User.NAME, user.getName());
+        doc.append(User.PASSWORD, user.getPassword());
+        doc.append(User.EMAIL, user.getEmail());
+        doc.append(User.ONLINE, user.getOnline());
+        String phone = user.getPhoneNumber();
+        if (phone != null && phone.length() > 0) {
+            doc.append(User.PHONE, phone);
+        }
+        db.getCollection(MyDatabase.USERS).insertOne(doc);
+    }
+
+    @Override
+    public List<User> findUsers(User filter) {
+        final List<User> findResult = new ArrayList<>();
+        try (MongoCursor<Document> cursor = db.getCollection(MyDatabase.USERS).find(userToBson(filter)).iterator()) {
+            while (cursor.hasNext()) {
+                findResult.add(docToUser(cursor.next()));
+            }
+        }
+        return findResult;
+
+    }
+
+    @Override
+    public User update(User user) {
+        Bson filter = eq(User.NAME, user.getName());
+        if (user.getPassword() != null) {
+            filter = and(filter, eq(User.PASSWORD, user.getPassword()));
+        }
+        final Document prevUserDoc = db.getCollection(MyDatabase.USERS).findOneAndUpdate(filter, new Document("$set", userToDoc(user)));
```

Comment:
As I mentioned in another thread, I still think we should clear username and password (if not null) before calling userToDoc().
I can give you an example that is not a directly proof that you should do this but somehow convinced myself.